### ISSUE TITLE
Getting "Could not load file or assembly Microsoft.Extensions.DependencyModel" in .NET 5.0

### ISSUE THREAD

**Issue by ld100** [dotnet/orleans#7172]
I've just built an Orleans-based app using examples from docs, but the project with Orleans client crashes with following error:

```
System.IO.FileLoadException: Could not load file or assembly 'Microsoft.Extensions.DependencyModel, Version=3.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The located assembly's manifest definition does not match the assembly reference. (0x80131040)\nFile name: 'Microsoft.Extensions.DependencyModel, Version=3.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'\n at Orleans.ApplicationPartManagerExtensions.AddFromDependencyContext
...
```
[dotnetbug.zip](https://github.com/dotnet/orleans/files/6899034/dotnetbug.zip)

I'm running the app as a Linux Docker container.
In the Gitter chat there was an assumption that something might be wrong with a way I package the app into container.

For the convenience I've published my *.csproj files and the Dockerfile [in the Gist](https://gist.github.com/ld100/fa5b3ae9f5f1be033a287b19896f7670).

*Keycard.Grains.Interfaces.csproj* contains grain interfaces
*Keycard.Grains.csproj* contains grains, just like in examples from docs
*Keycard.Worker.csproj* contains the Silo code co-hosted with the asp.net core.
*Keycard.Messenger.csproj* contains the Client code co-hosted with the asp.net core.

**Comment by ReubenBond** [comment]
Thanks for creating an issue, @ld100.

Could you try adding the following PackageReference to your client project and any other project which has this issue?

``` xml
<PackageReference Include="Microsoft.Extensions.DependencyModel" Version="5.0.0" />
```

I would have imagined that this version of the package should be pulled in automatically given you are using .NET 5.0, but seemingly not.

**Comment by ld100** [comment]
@ReubenBond thanks! The first solution didn't help. The second one looks quite magical to me, but it works :)

**Comment by ReubenBond** [comment]
Do you have Microsoft.Extensions.DependencyModel in your published output folder?

**Comment by ld100** [comment]
Yes, there is "Microsoft.Extensions.DependencyModel.dll" there.

**Comment by ReubenBond** [comment]
Fixed by #7197
